deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -703,7 +1026,98 @@ def get_message(self, event_type=None):
         return self.generate_message()
 
 
+class GeneratorHostinfo:
+    """This class allows the generation of hostinfo events.
+
+    Creates hostinfo events, randomizing an open port detection template event on a host.
+    It randomizes the host, as well as the ports and their protocol. The number of open ports of the event is a
+    random number from 1 to 10. Example of hostinfo message:
+
+        3:/var/log/nmap.log:Host: 95.211.24.108 (), open ports: 43270 (udp) 37146 (tcp) 19885 (tcp)
+    """
+    def __init__(self):
+        self.HOSTINFO_MQ = 3
+        self.hostinfo_basic_template = 'Host: <random_ip> (), open ports: '
+        self.protocols_list = ['udp', 'tcp']
+        self.localfile = '/var/log/nmap.log'
+
+    def generate_event(self):
+        number_open_ports = randint(1, 10)
+        host_ip = get_random_ip()
+        message_open_port_list = ''
+        for i in range(number_open_ports):
+            message_open_port_list += fr"{randint(1,65535)} ({choice(self.protocols_list)}) "
+
+        message = self.hostinfo_basic_template.replace('<random_ip>', host_ip)
+        message += message_open_port_list
+        message = fr"{self.HOSTINFO_MQ}:{self.localfile}:{message}"
+
+
+        return message
+
+
+class GeneratorWinevt:
+    """This class allows the generation of winevt events.
+
+    Create events of the different winevt channels: System, Security, Application, Windows-Defender and Sysmon.
+    It uses template events (`data/winevt.py`) for which the `EventID` field is randomized. Message structure:
+
+        f:EventChannel:{"Message":"<EVENTCHANNEL_MESSAGE>","Event":"<EVENT_CHANNEL_EVENT_XML>"}
+
+    Args:
+        agent_name (str): Name of the agent.
+        agent_id (str): ID of the agent.
+    """
+    def __init__(self, agent_name, agent_id):
+        self.agent_name = agent_name
+        self.agent_id = agent_id
+        self.WINENVT_MQ = 'f'
+        self.WINENVT = 'Eventchannel'
+        self.winevt_sources = [('system', winevt.WINEVT_SYSTEM), ('security', winevt.WINEVT_SECURITY),
+                               ('application', winevt.WINEVT_APPLICATION),
+                               ('windows-defender', winevt.WINEVT_WINDOWS_DEFENDER),
+                               ('sysmon', winevt.WINEVT_SYSMON)]
```

Comment:
This should be a dictionary, not a list of tuples.

Reply:
Sure. https://github.com/wazuh/wazuh-qa/pull/1179/commits/4614e3f1c54dfb77be82318af456b544785fe3a2